test/e2e/storage/volume_metrics.go

```diff
@@ -179,18 +177,26 @@ func getControllerStorageMetrics(ms metrics.ControllerManagerMetrics) map[string
 // Verifies the specified metrics are in `kubeletMetrics`
 func verifyVolumeStatMetric(metricKeyName string, namespace string, pvcName string, kubeletMetrics metrics.KubeletMetrics) {
 	found := false
+	invalidSamples := []string{}
```

Comment:
You're only using the len of the slice to detect if a sample error occurred, not the contents. And you log the bad sample so I don't think a slice of sample names is needed here. Perhaps just a counter, or another bool?
I'd delete `found`, change the invalid samples slice to `var errcnt`, increment errcnt each place you detect and error, and add a `Logf` to the case where the pvc name and/or ns don't match. When the `for` is done you can `Expect` errcnt to be zero.

Reply:
That works for the sample errors and I'll make that change. We still need `found` because there could be metrics from previous runs that have different PVC/namespace names. We want to ignore those and so can only discover we have an error after iterating through all the metrics.

Reply:
Made the above change